app/docker/views/swarm/visualizer/swarmVisualizerController.js

```diff
@@ -1,17 +1,25 @@
 angular.module('portainer.docker')
-.controller('SwarmVisualizerController', ['$q', '$scope', '$document', '$interval', 'NodeService', 'ServiceService', 'TaskService', 'Notifications',
-function ($q, $scope, $document, $interval, NodeService, ServiceService, TaskService, Notifications) {
+.controller('SwarmVisualizerController', ['$q', '$scope', '$document', '$interval', 'NodeService', 'ServiceService', 'TaskService', 'Notifications', 'LocalStorage',
+function ($q, $scope, $document, $interval, NodeService, ServiceService, TaskService, Notifications, LocalStorage) {
 
   $scope.state = {
-    ShowInformationPanel: true,
-    DisplayOnlyRunningTasks: false,
-    refreshRate: '5'
+    ShowInformationPanel: LocalStorage.getSwarmVisualizerSettings('show_info_panel', true),
+    DisplayOnlyRunningTasks: LocalStorage.getSwarmVisualizerSettings('display_only_running_tasks', false),
+    refreshRate: LocalStorage.getSwarmVisualizerSettings('refresh_rate', '5')
   };
 
   $scope.$on('$destroy', function() {
     stopRepeater();
   });
 
+  $scope.$watch('state.DisplayOnlyRunningTasks', function(newVal, oldVal) {
```

Comment:
As these changes are triggered by interaction with the DOM, I'd rather see the usage of `ng-change` functions here than the introduction of $watch functions. Could you replace these?